examples/informers/DeltaChangesQueryingController.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using k8s;
+using k8s.Informers;
+using k8s.Informers.Notifications;
+using k8s.Models;
+using KellermanSoftware.CompareNetObjects;
+using Microsoft.Extensions.Logging;
+
+namespace informers
+{
+    // this sample demos both informer and controller
+    // there are two loggers:
+    //   _informerLogger lets you see raw data coming out of informer stream
+    //   _reconcilerLogger lets you see batches of object transitions that object went through since last time we did work on it
+    // reconciler is purposely slowed down to show accumulation of events between worker actions
+
+    // try creating and deleting some pods in "default" namespace and watch the output
+
+    public class DeltaChangesQueryingController : IController
+    {
+        private readonly IKubernetesInformer<V1Pod> _podInformer;
+        private readonly CompositeDisposable _subscription = new CompositeDisposable();
+        private readonly ILogger _informerLogger;
+        private readonly CompareLogic _objectCompare = new CompareLogic();
+
+        public DeltaChangesQueryingController(IKubernetesInformer<V1Pod> podInformer, ILoggerFactory loggerFactory)
+        {
+            _podInformer = podInformer;
+            _informerLogger = loggerFactory.CreateLogger("Informer");
+            _objectCompare.Config.MaxDifferences = 100;
+        }
+
+
+        public Task Initialize(CancellationToken cancellationToken)
+        {
+            _podInformer
+                .GetResource(ResourceStreamType.ListWatch, KubernetesInformerOptions.Builder.NamespaceEquals("default").Build())
+                // .Resync(TimeSpan.FromSeconds(10))
+                .Catch<ResourceEvent<V1Pod>, Exception>(e =>
+                {
+                    _informerLogger.LogCritical(e, e.Message);
+                    return Observable.Throw<ResourceEvent<V1Pod>>(e);
+                })
+                .Buffer(TimeSpan.FromSeconds(5))
+                .Where(x => x.Any())
+                .Do(x =>
+                {
+                    var eventsPerResource = x.GroupBy(x => x.Value.Metadata.Name);
+                    foreach (var item in eventsPerResource)
+                    {
+                        Reconcile(item.ToList());
+                    }
+                })
+                .Subscribe()
+                .DisposeWith(_subscription);
+            return Task.CompletedTask;
+        }
+
+        private async void Reconcile(IList<ResourceEvent<V1Pod>> changes)
```

Comment:
async void makes me sad. Also what happens when these calls overlap?

Reply:
Async is an artifact of refactoring, this can just be void for this sample. I've updated sample. Given that I took out the code you didn't like that was on top of TPL, this is very primitive controller and not something you would use in production. Worker / reconciler type manager is going to be part of controller  building blocks which is something I still need to tackle. Those new blocks would ensure that we're only running a single reconcile process against a given resource and potentially support cancellation semantics (no point in finishing current reconciliation if the desired state has changed). For now, this is meant to highlight the informers piece.